### ISSUE TITLE
Nil returning/no return functions not executed when called by nil returning functions

### ISSUE THREAD

**Issue by MaryamZi** [ballerina-platform/ballerina-lang#11984]
**Description:**
The following does not result in a panic, returns nil instead.

```ballerina
import ballerina/io;

public function main() {
    io:println(retErOrNil());
}

function retErOrNil() returns error? {
    return panicTest();
}

function panicTest() {
    error e = error("test error");
    panic e;
}

```

**Comment by MaryamZi** [comment]
Seems to be a larger issue with nil returning functions invoking nil returning/no return functions.

Consider:
```ballerina
import ballerina/io;

public function main() {
    nilRetFunc();
}

function nilRetFunc() returns () {
    return noRetFunc();
}

function noRetFunc() {
    io:println("********** Executing the no return function **********");
}
```

Running the above doesn't result in the output "********** Executing the no return function **********".

`noRetFunc()` doesn't seem to be executed at all.

**Comment by SupunS** [comment]
Fixed with #12029
